compiler/rustc_mir/src/borrow_check/type_check/mod.rs

```diff
@@ -1281,37 +1282,41 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                             .eq(output_ty, revealed_ty)?,
                     );
 
-                    for (&opaque_def_id, opaque_decl) in &opaque_type_map {
+                    for &(opaque_type_key, opaque_decl) in &opaque_type_map {
                         let resolved_ty = infcx.resolve_vars_if_possible(opaque_decl.concrete_ty);
                         let concrete_is_opaque = if let ty::Opaque(def_id, _) = resolved_ty.kind() {
-                            *def_id == opaque_def_id
+                            *def_id == opaque_type_key.def_id
                         } else {
                             false
                         };
-                        let opaque_defn_ty = match concrete_opaque_types.get(&opaque_def_id) {
+
+                        let concrete_ty = match concrete_opaque_types
+                            .iter()
+                            .find(|(key, _)| key.def_id == opaque_type_key.def_id)
+                            .map(|(_, ty)| ty)
```

Comment:
This is exactly equal to the newly added fn `find_concrete_ty_from_def_id`, defined here https://github.com/rust-lang/rust/pull/86118/files#diff-388ddc23229cbd185e029e3c11c76b63c2ddafdd9495cca461ce075e73010a8dR807
I guess it would be nice if we reuse that somehow.